### PR TITLE
Fix sched slip

### PR DESCRIPTION
**Pull request by jho44**
- store dates in UTC
- hopefully prevents sched slippage now, if i understood [this](https://github.com/lilith/playdate/pull/48#issuecomment-1668560181) properly
- ensure that, when saving new reminderDatetime, it's not before `now`

### CODE REVIEW

File: src/lib/server/twilio.ts [hunk 1]
```diff
@@ -338,7 +338,7 @@ export async function sendNotif() {
 				await sendMsg({ phone, type: 'reminder' }, null);
 
 				// update reminder date for next notif -- x days from today
-				const newReminderDate = new Date(now);
+				const newReminderDate = new Date(reminderDatetime);
```
Comment (lilith) [inline src/lib/server/twilio.ts:341]: It looks like you're loading as a UTC date but saving here in local time.

**Comment by lilith** [comment]
Dates are extremely hard. I would suggest factoring out logic around dates into functions you can easily test to ensure they behave as you expect. Since you're only storing the reminder info in a single date field, any errors will compound. 

When saving the user profile, you can combine all the fields into a date time in the user's time zone pretty easily. But later, you need to be very precise about changes, because the user will cross a DST boundary twice yearly. You can't just add days because you actually need to add 23 or 25 hours if you're crossing a boundary. Fortunately most sane date libraries handle this appropriately if you have a date value with time zone information.

The way I would suggest adding days is to read both the user time zone and the reminderDate (stored as UTC), then convert it to the user's local time. Then extract the time component in the user's time zone. Create a new date in the user's time zone at the appropriate future interval, then set the time component to match the original value. Then convert that to UTC and store. That way, the UTC conversion doesn't mess with the time, and you're not storing local time in the db.

File: src/lib/server/twilio.ts (hunk 1 above)
Comment (jho44) [reply src/lib/server/twilio.ts:332]: you're right, sorry x.x now that we're not saving reminderDatetime in the user's local time, we can get rid of all timezone logic in the reminder func. pushed a fix~

**Comment by jho44** [comment]
> For the cron, the way I would suggest adding days is to read both the user time zone and the reminderDate (stored as UTC), then convert it to the user's local time. Then extract the time component in the user's time zone. Create a new date in the user's time zone at the appropriate future date w/ interval, then set the time component on that date to match the original value. Then convert that to UTC and store. That way, the UTC conversion doesn't mess with the time, and you're not storing local time in the db. Since the UTC offset depends on the calendar date.

umm will give this a go

**Comment by jho44** [comment]
@lilith hopefully I was able to get things right 👀 i put some common date conversion funcs in `lib/server/date.ts`. before getting new reminder dates, i convert them into the user's local time, add the interval, set the original hr + mins, then turn back into UTC. saving the reminder date in the Profile page works as expected (i.e., if you save as 4:30PM, it'll show as 4:30PM even after refreshing tho it's saved to the db in UTC). and a reminder text is triggered at the right time (wrt the user's timezone)

i tried testing with DST crossing.
`getNewReminderDate(new Date('11/4/2023 5:00:00 PM PDT'), 'America/Los_Angeles', 2)` says the new reminder date is:
1. in local time -- 11/6/2023 5PM
2. in UTC -- 11/7/2023 1AM

Screenshot 2023-08-07 at 9 09 41 PM

**Pull request merged**